src/utils/i18n/i18nextLocize.ts

```diff
@@ -324,7 +328,7 @@ export const fetchTranslations = async (lang: string): Promise<I18nextResources>
  * @param lang
  * @param defaultLocales
  */
-const i18nextLocize = (lang: string, defaultLocales: I18nextResources): i18n => {
+const createI18nextLocizeInstance = (lang: string, i18nTranslations: I18nextResources): i18n => {
```

Comment:
Good catch, maybe memoization would be a better fix to this, I'm a fan of functional programming (immutability/pure functions/etc), that's just my personal opinion though.

See: https://www.npmjs.com/package/memoize-one

Reply:
What advantages would memoization have over a singleton design in this case?

It's an interesting alternative, I'd just like to see pros/cons before jumping on it.

Reply:
The advantage of this approach is that it's a single function, you don't need to rely on anything other than the function itself, since it's a pure function.